File: server/src/main/java/au/com/codeka/warworlds/server/admin/handlers/AjaxDesignsHandler.java
package au.com.codeka.warworlds.server.admin.handlers;

import au.com.codeka.warworlds.common.proto.Designs;
import au.com.codeka.warworlds.common.sim.DesignHelper;
import au.com.codeka.warworlds.server.handlers.RequestException;

/**
 * Handler for /admin/ajax/designs, returning the list of ship/building designs as a JSON file.
 */
public class AjaxDesignsHandler extends AdminHandler {
  @Override
  public void get() throws RequestException {
    setResponseJson(new Designs.Builder().designs(DesignHelper.getDesigns()).build());
  }
}

File: server/src/main/java/au/com/codeka/warworlds/server/admin/handlers/AjaxDesignsHandler.kt
package au.com.codeka.warworlds.server.admin.handlers

import au.com.codeka.warworlds.common.proto.Designs
import au.com.codeka.warworlds.common.sim.DesignHelper
import au.com.codeka.warworlds.server.handlers.RequestException

/**
 * Handler for /admin/ajax/designs, returning the list of ship/building designs as a JSON file.
 */
class AjaxDesignsHandler : AdminHandler() {
  @Throws(RequestException::class)
  public override fun get() {
    setResponseJson(Designs.Builder().designs(DesignHelper.designs).build())
  }
}